homeassistant/components/paradox.py

```diff
@@ -0,0 +1,282 @@
+    _prt_speed = config.get(CONF_PRT_SPEED)
+
+    # Needed for the IP100/150
+    # _ip_host = config.get(CONF_IP_HOST)
+    # _ip_port = config.get(CONF_IP_PORT)
+    # _zone_dump = config.get(CONF_ZONEDUMP_INTERVAL)
+    # _keep_alive = config.get(CONF_IP_KEEPALIVE)
+    # Assign it here or rather send it all with kwargs?
+    # How do we need to communicate with alarm panel
+    # if 'PRT3' in _comm_module:
+    #    PRT_MODULE = True
+
+    # if 'IP' in _comm_module:
+    #    IP_MODULE = True
+
+    # These will be auto discovered in later versions.
+    # ...if we can get the serial speed sorted.
+    _zones = config.get(CONF_ZONES)
+    _partitions = config.get(CONF_PARTITIONS)
+    # _connect_status = {}
+
+    if PARADOX_CONTROLLER is None:
+        _LOGGER.info('Setting up %s on port %s.', _paradox_model, _prt_port)
+        # Rather pass all the parameters using kwargs?
+        PARADOX_CONTROLLER = ParadoxAlarmPanel(_paradox_model,
+                                               _comm_module,
+                                               # _user,
+                                               # _pass,
+                                               _prt_port,
+                                               _prt_speed)
+        if PARADOX_CONTROLLER is None:
+            _LOGGER.info('Paradox controller not initialised')
+            return False
+        else:
+            _LOGGER.info('Paradox controller initialised as %s.',
+                         PARADOX_CONTROLLER.paradox_model)
+
+    def update_alarm_armed_cb(area_number):
+        """Process the area armed event received from alarm panel."""
+        _LOGGER.debug('Area %d received armed event.', area_number)
+        try:
+            # Area in use on alarm panel might not be setup/defined in HA
+            _affected_area = PARTITION_SCHEMA(_partitions[area_number])
+            _LOGGER.debug('HA area %s to be armed.',
+                          _affected_area[CONF_PARTITIONNAME])
+            # This does not seem to be the correct way to set the state.
+
+            _att = {'friendly_name': _affected_area[CONF_PARTITIONNAME]}
+
+            hass.states.set('alarm_control_panel.' +
+                            _affected_area[CONF_PARTITIONNAME],
+                            STATE_ALARM_ARMED_AWAY, _att)
+        except KeyError:
+            _LOGGER.debug('Area %d not defined in HA.', area_number)
+
+        return True
+
+    def update_alarm_stay_armed_cb(area_number):
+        """Process area stay armed event received from alarm panel."""
+        _LOGGER.debug('Area %d received stay armed event.', area_number)
+        try:
+            # Area in use on alarm panel might not be setup/defined in HA
+            _affected_area = PARTITION_SCHEMA(_partitions[area_number])
+            _LOGGER.debug('HA area %s to be stay armed.',
+                          _affected_area[CONF_PARTITIONNAME])
+            # This does not seem to be the correct way to set the state.
```

Comment:
Correct 😉 

Platform entities have to subscribe to the paradox hub itself and update when necessary.